File: model/Constraint.vb
Public Class Constraint
    Public Table As Table
    Public Name As String
    Public Type As String
    Public Clustered As Boolean
    Public Columns As New List(Of String)

    Public ReadOnly Property ClusteredText() As String
        Get
            If Not Clustered Then Return ""
            Return "CLUSTERED"
        End Get
    End Property

    Public Sub New(ByVal name As String, ByVal type As String, Optional ByVal columns As String = "")
        Me.Name = name
        Me.Type = type
        If Not String.IsNullOrEmpty(columns) Then
            Me.Columns = New List(Of String)(columns.Split(","c))
        End If
    End Sub

    Public Function Script() As String
        If Type = "INDEX" Then
            Return String.Format("CREATE {0} INDEX [{1}] ON [{2}].[{3}] ([{4}])", _
                   ClusteredText, Name, Table.Owner, Table.Name, String.Join("], [", Columns.ToArray()))
        End If
        Return String.Format("CONSTRAINT [{0}] {1} {2} ([{3}])", _
                            Name, Type, ClusteredText, String.Join("], [", Columns.ToArray()))
    End Function
End Class

File: model/Constraint.cs
using System.Collections.Generic;

namespace SchemaZen.model {
	public class Constraint : INameable, IScriptable {
		public bool Clustered;
		public List<string> Columns = new List<string>();
		public List<string> IncludedColumns = new List<string>();
		public string Name { get; set; }
		public Table Table;
		public string Type;
		public bool Unique;
		private bool IsNotForReplication;
		private string CheckConstraintExpression;

		public Constraint(string name, string type, string columns) {
			Name = name;
			Type = type;
			if (!string.IsNullOrEmpty(columns)) {
				Columns = new List<string>(columns.Split(','));
			}
		}

		public static Constraint CreateCheckedConstraint(string name, bool isNotForReplication, string checkConstraintExpression) {
			var constraint = new Constraint(name, "CHECK", "") {
				IsNotForReplication = isNotForReplication,
				CheckConstraintExpression = checkConstraintExpression
			};
			return constraint;
		}

		public string ClusteredText {
			get { return !Clustered ? "NONCLUSTERED" : "CLUSTERED"; }
		}

		public string UniqueText {
			get { return Type != "PRIMARY KEY" && !Unique ? "" : "UNIQUE"; }
		}

		public string ScriptCreate() {
			if (Type == "CHECK") {
				return $"CONSTRAINT [{Name}] CHECK {CheckConstraintExpression}";
			}

			if (Type == "INDEX") {
				var sql = string.Format("CREATE {0} {1} INDEX [{2}] ON [{3}].[{4}] ([{5}])", UniqueText, ClusteredText, Name,
					Table.Owner, Table.Name,
					string.Join("], [", Columns.ToArray()));
				if (IncludedColumns.Count > 0) {
					sql += string.Format(" INCLUDE ([{0}])", string.Join("], [", IncludedColumns.ToArray()));
				}
				return sql;
			}
			return (Table.IsType ? string.Empty : string.Format("CONSTRAINT [{0}] ", Name)) +
				string.Format("{0} {1} ([{2}])", Type, ClusteredText, string.Join("], [", Columns.ToArray()));
		}
	}
}
